### PR TITLE
[api] EventNotifyBackendJob now uses ActiveJob

### PR DESCRIPTION
**Pull request by evanrolfe**
Part of this card: https://trello.com/c/Tx9d3bUb/467-5-p4-event-system-change-delayed-jobs-to-be-created-with-activejobs

### CODE REVIEW

File: src/api/spec/jobs/event_notify_backend_job_spec.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+RSpec.describe EventNotifyBackendJob, type: :job, vcr: true do
+  include ActiveJob::TestHelper
+
+  describe '#perform' do
+    let!(:project) { create(:project) }
+    let(:response) { double(:response, body: '<response code="ok"/>') }
+
+    before do
+      Event::CreateProject.create(project: project.name)
+      allow(Backend::Connection).to receive(:post).and_return(response)
```
Comment (ChrisBr) [inline src/api/spec/jobs/event_notify_backend_job_spec.rb]: What happens if response is not ``ok`` (or an exception?)

File: src/api/app/jobs/event_notify_backend_job.rb [hunk 2]
```diff
@@ -0,0 +1,5 @@
+class EventNotifyBackendJob < ApplicationJob
+  def perform
+    Event::Base.not_in_queue.find_each(&:notify_backend)
```
Comment (ChrisBr) [inline src/api/app/jobs/event_notify_backend_job.rb]: No queue_as necessary?

Comment (evanrolfe) [inline src/api/app/jobs/event_notify_backend_job.rb]: No, this job will perform in the "default" queue by default.

File: src/api/spec/jobs/event_notify_backend_job_spec.rb (hunk 1 above)
Comment (evanrolfe) [inline src/api/spec/jobs/event_notify_backend_job_spec.rb]: The same thing that happened before this change. The event is marked as `queued: true` before it tries to notify the backend, so if that fails then the backend will never be notified. This needs to be fixed but this is part of the bigger changes we are making to the events system.

**Pull request merged**